lib/Doctrine/DataDict/Mysql.php

```diff
@@ -352,7 +351,7 @@ public function getPortableDeclaration(array $field)
                         $value  = str_replace('\'\'', '\'', $value);
                         $length = max($length, strlen($value));
                     }
-                    if ($length == '1' && count($matches[1]) == 2) {
+                    if ($dbType === 'enum' && $length == '1' && count($matches[1]) == 2) {
```

Comment:
Except SET won't be treated as a boolean since it allows empty string and multiple values.